http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -240,6 +254,19 @@ class CollectorService[F[_]: Sync](
       }
     }
 
+  /**
+    * Creates an Access-Control-Allow-Origin header which specifically allows the domain which made
+    * the request
+    *
+    * @param request Incoming request
+    * @return Header allowing only the domain which made the request or everything
+    */
+  def accessControlAllowOriginHeader(request: Request[F]): Header.Raw =
+    Header.Raw(
+      ci"Access-Control-Allow-Origin",
+      extractHostsFromOrigin(request.headers).headOption.map(_.renderString).getOrElse("*")
```

Comment:
Shouldn't we put the whole list instead of just one with `headOption` ?

Reply:
[Only a single origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin) can be specified with `Access-Control-Allow-Origin`.